src/js/components/Carousel/__tests__/Carousel-test.tsx

```diff
@@ -1,139 +1,157 @@
-      <Grommet>
-        <Carousel data-testid="test-carousel">
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
-        </Carousel>
-      </Grommet>,
+  test('arrow navigation: next', async () => {
+    render(
+      <Carousel controls="arrows">
+        <div>Slide One</div>
+        <div>Slide Two</div>
+        <div>Slide Three</div>
+      </Carousel>,
     );
-    expect(container.firstChild).toMatchSnapshot();
-
-    fireEvent.keyDown(getByTestId('test-carousel'), {
-      key: 'Right',
-      keyCode: 39,
-      which: 39,
-    });
-    expect(container.firstChild).toMatchSnapshot();
 
-    fireEvent.keyDown(getByTestId('test-carousel'), {
-      key: 'Left',
-      keyCode: 37,
-      which: 37,
+    /**
+     * - Currently on "Slide One"
+     * - Simulating click on the next arrow button
+     * - Expecting "Slide Two" to be visible
+     */
+    const nextButton = screen.getByLabelText('Next');
+    userEvent.click(nextButton);
+    await waitFor(() => {
+      expect(getSlideOne()).not.toBeVisible();
+      expect(getSlideTwo()).toBeVisible();
     });
-    expect(container.firstChild).toMatchSnapshot();
   });
 
-  test('should trigger events of focus, blur and click', () => {
-    const onFocus = jest.fn();
-    const onBlur = jest.fn();
-    const { container } = render(
-      <Grommet>
-        <Carousel
-          id="test-carousel"
-          onFocus={onFocus}
-          onBlur={onBlur}
-          controls="selectors"
-        >
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
-        </Carousel>
-      </Grommet>,
+  test('arrow navigation: previous', async () => {
+    render(
+      <Carousel controls="arrows">
+        <div>Slide One</div>
+        <div>Slide Two</div>
+        <div>Slide Three</div>
+      </Carousel>,
     );
 
-    const button = document
-      .getElementById('test-carousel')!
-      .querySelector('button');
+    /**
+     * - Currently on "Slide One"
+     * - Simulating click on the previous arrow button
+     * - Expecting "Slide Three" to be visible
+     */
+    const previousButton = screen.getByLabelText('Previous');
+    userEvent.click(previousButton);
+    await waitFor(() => {
+      expect(getSlideOne()).not.toBeVisible();
+      expect(getSlideThree()).toBeVisible();
+    });
+  });
 
-    expect(button).not.toBeNull();
+  test('selector navigation: forward', async () => {
+    render(
+      <Carousel controls="selectors">
+        <div>Slide One</div>
+        <div>Slide Two</div>
+        <div>Slide Three</div>
+      </Carousel>,
+    );
 
-    fireEvent.focus(button!);
-    expect(onFocus).toHaveBeenCalledTimes(1);
+    /**
+     * - Currently on "Slide One"
+     * - Simulating click on the 3rd selector button
+     * - Expecting "Slide Three" to be visible
+     */
+    const thirdSelector = screen.getAllByRole('button')[2];
```

Comment:
If we add a11yTitles to the selectors, we can be more explicit and select by the label text instead.